cogs/apply.py

```diff
@@ -1,18 +1,90 @@
 import sys
 
+from cogs.exceptions import ApplyError
 from cogs.helpers import *
+from gspread_formatting import BooleanCondition
+
+
+# data validation conditions
+conditions = [
+    "NUMBER_GREATER" "NUMBER_GREATER_THAN_EQ",
+    "NUMBER_LESS",
+    "NUMBER_LESS_THAN_EQ",
+    "NUMBER_EQ",
+    "NUMBER_NOT_EQ",
+    "NUMBER_BETWEEN",
+    "NUMBER_NOT_BETWEEN",
+    "TEXT_CONTAINS",
+    "TEXT_NOT_CONTAINS",
+    "TEXT_STARTS_WITH",
+    "TEXT_ENDS_WITH",
+    "TEXT_EQ",
+    "TEXT_IS_EMAIL",
+    "TEXT_IS_URL",
+    "DATE_EQ",
+    "DATE_BEFORE",
+    "DATE_AFTER",
+    "DATE_ON_OR_BEFORE",
+    "DATE_ON_OR_AFTER",
+    "DATE_BETWEEN",
+    "DATE_NOT_BETWEEN",
+    "DATE_IS_VALID",
+    "ONE_OF_RANGE",
+    "ONE_OF_LIST",
+    "BLANK",
+    "NOT_BLANK",
+    "CUSTOM_FORMULA",
+    "BOOLEAN",
+]
+
+# expected headers for different tables
+data_validation_headers = ["table", "range", "condition", "value"]
+problems_headers = [
+    "ID",
+    "table",
+    "cell",
+    "level",
+    "rule ID",
+    "rule name",
+    "value",
+    "fix",
+    "instructions",
+]
 
 
 def msg():
-    return "Apply a standardized problems table to the spreadsheet"
+    return "Apply a table to the spreadsheet"
 
 
-def apply(args):
-    """Apply a standardized problems table to the spreadsheet."""
-    validate_cogs_project()
-    set_logging(args.verbose)
+def apply_data_validation(data_valiation_tables):
+    """Apply one or more data validation rules to the sheets."""
     tracked_sheets = get_tracked_sheets()
+    add_dv_rules = {}
+    for data_validation_table in data_valiation_tables:
+        add_rows = []
+        for row in data_validation_table:
+            sheet_title = row["table"]
+            if sheet_title not in tracked_sheets.keys():
+                raise ApplyError(f"'{sheet_title}' is not a tracked sheet")
+            rule = clean_rule(row["table"], row["range"], row["condition"], row["value"])
+            add_rows.append(rule)
 
+        for row in add_rows:
+            sheet_title = row["Sheet Title"]
```

Comment:
I don't get this. Is the column named "table" or "Sheet Title"?

Reply:
It's "table" in the sheet you pass to "apply" - this is what we decided on for the ROBOT standardized problems tables so I'm following that pattern.

It's "Sheet Title" in the `.cogs/validation.tsv` file because that's the pattern we've used for all the other COGS data files.

It's annoying that it's not the same, but I also think it would be confusing to have inconsistencies between the sets of files.